src/fsharp/Optimizer.fs

```diff
@@ -3055,7 +3062,7 @@ and OptimizeBinding cenv isRec env (TBind(v,e,spBind)) =
             then {einfo with Info=UnknownValue} 
             else einfo 
         if v.MustInline  && IsPartialExprVal einfo.Info then 
-            errorR(InternalError("the mustinline value '"^v.LogicalName^"' was not inferred to have a known value",v.Range));
+            errorR(NumberedError(FSComp.SR.optValueMarkedInlineButIncomplete(v.DisplayName), v.Range))
```

Comment:
I'm not entirely sure that this only happens when `valu` (at line 2749) is set to `UnknownValue`,